gcm_filters/filter.py

```diff
@@ -60,8 +58,27 @@ class FilterSpec(NamedTuple):
 
 
 def _compute_filter_spec(
-    filter_scale, dx_min, n_steps, filter_shape, transition_width, root_tolerance=1e-12
+    filter_scale,
+    dx_min,
+    filter_shape,
+    transition_width,
+    ndim,
+    n_steps=0,
+    root_tolerance=1e-12,
 ):
+    # First set number of steps if not supplied by user
+    if n_steps == 0:
+        if filter_shape == FilterShape.GAUSSIAN:
+            if ndim == 1:
+                n_steps = np.ceil(1.3 * filter_scale / dx_min).astype(int)
+            else:  # ndim==2
+                n_steps = np.ceil(1.8 * filter_scale / dx_min).astype(int)
+        else:  # Taper
+            if ndim == 1:
+                n_steps = np.ceil(4.5 * filter_scale / dx_min).astype(int)
+            else:  # ndim==2
+                n_steps = np.ceil(6.4 * filter_scale / dx_min).astype(int)
```

Comment:
What if the user wants to do 3D `(ndim == 3)`? Is it not supported? Currently if someone passes `ndim=3` they will get a pretty undecipherable error.

Reply:
The only part not supported would be setting a default `n_steps`. I could add an error that you have to supply your own `n_steps` if you want to use `ndim` other than 1 or 2.